pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -194,14 +198,31 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
+	prometheusCertSecret, err := utils.ValidateCertPair(r.Client,
+		common.OperatorNamespace(),
+		render.PrometheusTLSSecretName,
+		"", // We don't need the key.
+		corev1.TLSCertKey,
+	)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("failed to retrieve %s", render.PrometheusTLSSecretName))
+		r.status.SetDegraded(fmt.Sprintf("Failed to retrieve %s", render.PrometheusTLSSecretName), err.Error())
+		return reconcile.Result{}, err
+	} else if prometheusCertSecret == nil {
```

Comment:
This now forces the user to setup monitoring before the tunnel will be set up. I'm not sure if we should have that as a pre requisite of the tunnel setup, as I would think not having monitoring setup should be a valid configuration.

You might want to double check with PM on this.

Reply:
I removed the dependency.